src/diffusers/models/transformers/latte_transformer_3d.py

```diff
@@ -165,6 +165,48 @@ def __init__(
     def _set_gradient_checkpointing(self, module, value=False):
         self.gradient_checkpointing = value
 
+    # Copied from diffusers.models.unets.unet_3d_condition.UNet3DConditionModel.enable_forward_chunking
+    def enable_forward_chunking(self, chunk_size: Optional[int] = None, dim: int = 0) -> None:
```

Comment:
I assume Latte doesn't use a novel transformer block like SD3? 
https://github.com/huggingface/diffusers/blob/973a62d4085fd2ceb8240c53419aee828be3557d/src/diffusers/models/transformers/transformer_sd3.py#L97

If so, we need to do it like this:

https://github.com/huggingface/diffusers/blob/973a62d4085fd2ceb8240c53419aee828be3557d/src/diffusers/models/attention.py#L182

Reply:
Latte uses the [`BasicTransformerBlock`](https://github.com/huggingface/diffusers/blob/bbd2f9d4e9ae70b04fedf65903fd1fb035437db4/src/diffusers/models/transformers/latte_transformer_3d.py#L110) which already has the chunked ff implementation here: https://github.com/huggingface/diffusers/blob/bbd2f9d4e9ae70b04fedf65903fd1fb035437db4/src/diffusers/models/attention.py#L515

Have I missed anything else that might be causing very poor memory improvements?

Reply:
As in, we close this PR without merging it, right? Since improvements are negligible for way too much increase in time required

Reply:
We might benefit from adding vae slicing and tiling support it here btw. Decode memory goes up to 19 GB

Reply:
Might be worth checking that, yes. 

This forward chunking phenomenon (less memory savings) is becoming evident for models that are purely transformer based as opposed to hybrid architectures (such as the SDXL UNet). For I2VXLGen or SVD, we see savings however, for pure transformer ones like Latte, SD3, we don’t. 

For the SD3 800M variant, I did see nice improvements in memory but not so much for the 2B one. So, I think there is some specific FLOP/param count pattern where chunking tends to shine better. 

We should investigate this phenomenon further because ff chunking could be crucial for these models. WDYT?

Reply:
Oh interesting, I haven't really explored much of the transfomer-based image/video models and can't comment much on why we see this behaviour, but we can definitely try and investigate. I will try and put together a script over the weekend unless we have something ready already.

Quick question: IIUC, we do chunking only on the FeedForward modules and never on the individual linear layers (such as qkv projection layers of attention). Isn't the benefit of doing chunking for ffn lost if we are never going to also apply it on ALL linear layers that follow? Apologies if I'm being thick-headed but it's been a while since I've looked at some of the internal modeling code and I can't wrap my head around this. We have Attn followed by FFN. In Attn, we don't do chunking, and we materialize the QK product, which should be a main contributor to overall memory required. In FFN, we do chunking and save some memory, but as I understand, this saving is largely "invisible" in the overall memory measurement, where we care about the total/max usage, due to attention projection layers (since it does not use chunking) and QK, right?

Reply:
We cannot do chunking on all linear layers just like that. Read more here:
https://huggingface.co/blog/reformer

Regarding QK materialization ping pong, you are likely disregarding the fact that attention computation is handled by SDPA which optimizes the memory bits already. So, I don’t think that is as much of an issue.